src/main/java/com/dao/UserDAO.java

```diff
@@ -0,0 +1,205 @@
+
+        } catch (SQLException e) {
+
+            e.printStackTrace();
+        }
+
+        return isPresent;
+    }
+
+    public User getByName(String userName) {
+
+        User user = null;
+
+        try (Connection connection = dataSource.getConnection()) {
+
+            PreparedStatement sql = connection.prepareStatement(GET_BY_NAME_QUERY);
+            sql.setString(1, userName);
+
+            ResultSet rs = sql.executeQuery();
+
+            if (rs.next())
+                user = createUser(rs);
+
+        } catch (SQLException e) {
+
+            e.printStackTrace();
+        }
+
+        return user;
+    }
+
+    public User add(String userName, String password, boolean isAdmin) {
+
+        //in service class check if user exists
+
+        User user = null;
+
+        try (Connection connection = dataSource.getConnection()) {
+
+            PreparedStatement sql = connection.prepareStatement(ADD_QUERY);
+            sql.setString(1, userName);
+            sql.setString(2, password);
+            sql.setBoolean(3, isAdmin);
+
+            sql.executeUpdate();
+
+            user = getByName(userName);
+
+        } catch (SQLException e) {
+
+            e.printStackTrace();
+        }
+
+        return user;
+    }
+
+    public boolean remove(String userName) {
+
+        //in service class check if user exists or admin
+
+        boolean isRemoved = false;
+
+        try (Connection connection = dataSource.getConnection()) {
+
+            PreparedStatement sql = connection.prepareStatement(REMOVE_QUERY);
+            sql.setString(1, userName);
+
+            sql.executeUpdate();
+
+            isRemoved = true;
+
+        } catch (SQLException e) {
+
+            e.printStackTrace();
+        }
+
+        return isRemoved;
+    }
+
+    public boolean update(String userName, String password, boolean isAdmin) {
+
+        //in service class check if user exists
+
+        boolean isUpdated = false;
+
+        try (Connection connection = dataSource.getConnection()) {
+
+            PreparedStatement sql = connection.prepareStatement(UPDATE_QUERY);
+            sql.setBoolean(1, isAdmin);
+            sql.setString(2, password);
+            sql.setString(3, userName);
+
+            sql.executeUpdate();
+
+            isUpdated = true;
+
+        } catch (SQLException e) {
+
+            e.printStackTrace();
+        }
+
+        return isUpdated;
+    }
+
+    public List<User> getAll() {
+
+        List<User> users = new ArrayList<>();
+
+        try (Connection connection = dataSource.getConnection()) {
+
+            PreparedStatement sql = connection.prepareStatement(GET_ALL_QUERY);
+
+            ResultSet rs = sql.executeQuery();
+
+            while (rs.next())
+                users.add(createUser(rs));
+
+        } catch (SQLException e) {
+
+            e.printStackTrace();
+        }
+
+        return users;
+    }
+
+    private User createUser(ResultSet rs) throws SQLException {
```

Comment:
I think it would be better to change name of this method to smth like parseUser or createUserEntity, because createUser sounds quite ambiguous to me